config/init.go

```diff
@@ -14,46 +29,62 @@ type Config struct {
 	DockerContext      string
 	PublicKeyPEMBytes  []byte
 	InitialHash        string
-	TokenWindow        uint16
+	DataDir            string
+	TokenWindow        time.Duration
+	DeployInterval     time.Duration
 }
 
 var AppConfig Config
 
 func init() {
 	viper.SetEnvPrefix("CD")
 	viper.AutomaticEnv()
-	viper.SetDefault("COMPOSE_FILE", "/run/secrets/compose-file")
-	viper.SetDefault("COMPOSE_PROJECT_NAME", "")
-	viper.SetDefault("DOCKER_CONTEXT", "default")
-	viper.SetDefault("PUBKEY_FILE", "./keypairs/ecpubkey.pem")
-	viper.SetDefault("INITIAL_HASH", "f8c0c5c0811c1344e6948c5fabc2839151cd7f0444c2724f2cddd238ce62bdec")
-	viper.SetDefault("TOKEN_WINDOW", 60)
-	viper.SetDefault("BINDING", ":3000")
+	viper.SetDefault(envComposeFile, "/run/secrets/compose-file")
+	viper.SetDefault(envComposeProjectName, "")
+	viper.SetDefault(envDockerContext, "default")
+	viper.SetDefault(envPubkeyFile, "./keypairs/ecpubkey.pem")
+	viper.SetDefault(envInitialHash, "f8c0c5c0811c1344e6948c5fabc2839151cd7f0444c2724f2cddd238ce62bdec")
+	viper.SetDefault(envTokenWindow, 60)
+	viper.SetDefault(envPortBinding, ":3000")
+	viper.SetDefault(envDeployInterval, 20)
+	viper.SetDefault(envDatadir, "./data/")
 }
 
 func init() {
-	workingDir := viper.GetString("HOST_COMPOSE_WORKING_DIR")
+	workingDir := viper.GetString(envHostComposeWorkingDir)
 	if len(workingDir) == 0 {
-		panic("ENV: CD_HOST_COMPOSE_WORKING_DIR is not configured")
+		panic("ENV: CD_" + envHostComposeWorkingDir + " is not configured")
 	}
 
 	// 6071  openssl genpkey -algorithm EC -out eckey.pem \
```

Comment:
please review a public/private key generation command here if it's secured enough